hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/ddl/HiveSyncMode.java

```diff
@@ -26,6 +26,10 @@
    * The HMS mode use the hive meta client to sync metadata.
    */
   HMS,
+  /**
+   * The GLUE mode use the glue client to sync metadata.
+   */
+  GLUE,
```

Comment:
Not sure if `GLUE` belongs to kind of sync mode ? Even though GLUE uses supports hive protocol, we may need to think how to re-organize these syncing params.

Reply:
> Not sure if `GLUE` belongs to kind of sync mode ? Even though GLUE uses supports hive protocol, we may need to think how to re-organize these syncing params.


The glue client depends on hiveconf, and the parameters currently used in my test are configured as follows:

  'hive_sync.enable' = 'true',
  'hive_sync.mode' = 'glue',
  'hive_sync.table' = 'test_hudi_glue',
  'hive_sync.skip_ro_suffix' = 'true',
  'hive_sync.db' = 'hudi_meta_glue'

Reply:
@todd5167 this is not the right category for sync mode. sync mode only applies to hive sync tool, which uses jdbc/hiveql/hms. To use Glue, people need to configure sync tool class to `AWSGlueCatalogSyncTool`

Reply:
The way of configuring glue sync tool should be fixed on flink side; instead of setting mode, it should be setting the sync tool class. Changing the enum confused users who were misled to set it but no effect when using spark.

@todd5167 can you make a fix for this?

Reply:
> The way of configuring glue sync tool should be fixed on flink side; instead of setting mode, it should be setting the sync tool class. Changing the enum confused users who were misled to set it but no effect when using spark.
> 
> @todd5167 can you make a fix for this?

@xushiyan  In the future, I will change the synchronization Glue judgment condition to judgment through CatalogSyncToolClass, and the HiveSyncMode is still HMS.

Reply:
Thanks!

Reply:
@xushiyan @danny0405   cc https://github.com/apache/hudi/pull/5909/files